### PR TITLE
feat: handle quoteError as form state

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Agreed with product to use these copies, and precision 6 so we don't get overflows.
~~Note, the copies haven't been approved by @shapeshift/product yet, and the precision issue needs to be tackled, as we can't always fit the minimum amount in the preview button~~

This 

- adds a `quoteError` field to the swapper form state
- rewords the `amountTooSmall` and `sellAmountDoesNotCoverFee` copies to fit into a button
- handles the `SwapErrorTypes.TRADE_QUOTE_AMOUNT_TOO_SMALL` and  `SwapErrorTypes.TRADE_QUOTE_INPUT_LOWER_THAN_FEES` errors in the swapper input state (disabled/error state) vs. on preview button click currently
- revamps the logic so that toasts are not displayed for these errors (not 100% accurate because of the ramifications of swapper, expect generic toasts to still pop up)

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/pull/2403 

## Risk

- The error handling has been slightly revamped and we could be displaying errors twice (as swapper input state and as a toast)

## Testing (enable the `CowSwap` flag)

- Test swapping an amount that would equal in a value lower than fees e.g 10 VBNT -> FOX
- The swapper should be in a "Sell amount lower than fees" errored state, without the possibility to continue

- Test swapping an amount that is less than the minimum for that pair e.g 19 USDC -> FOX
- The swapper should be in a "Minimum: 20 USDC" errored state, without the possibility to continue

- Test swapping an amount that is matching the minimum for that pair e.g 20 USDC -> FOX
- The swapper should not be in an errored state and allow continuing with the swap

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2421** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2421" target="_blank">Graphite</a> 
    * **PR #2422** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2422" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2422?utm_source=stack-comment).

**Comment by willyogo** [comment]
This looks awesome @gomesalexandre 🔥  

re: overflowing amounts - let's truncate to 6 decimal places

File: src/components/Trade/TradeInput.tsx [hunk 1]
```diff
@@ -352,7 +350,7 @@ export const TradeInput = ({ history }: RouterProps) => {
                 aria-label='Switch'
                 isRound
                 icon={<FaArrowsAltV />}
-                isLoading={!!!quote}
```
Comment (0xApotheosis) [inline src/components/Trade/TradeInput.tsx:355]: Ah yes, the ol' triple exclamation operator.

File: src/components/Trade/TradeInput.tsx [hunk 2]
```diff
@@ -220,6 +206,18 @@ export const TradeInput = ({ history }: RouterProps) => {
       return 'common.insufficientAmountForGas'
     }
 
+    if (Boolean(quoteError)) {
```
Comment (0xApotheosis) [inline src/components/Trade/TradeInput.tsx:209]: Out of curiosity @gomesalexandre is there any reason you prefer `Boolean` over `!!`?

**Review by 0xApotheosis — APPROVED** [review]
Does what it says on the box 👌

**Pull request merged**